app/models/concerns/hostext/operating_system.rb

```diff
@@ -0,0 +1,55 @@
+module Hostext
+  module OperatingSystem
+    extend ActiveSupport::Concern
+    included do
+      scope :with_os, -> { where('hosts.operatingsystem_id IS NOT NULL') }
+      alias_attribute :os, :operatingsystem
+      validates :operatingsystem_id, :presence => true,
+        :if => ->(host) { host.managed }
+    end
+
+    # returns a configuration template (such as kickstart) to a given host
+    def provisioning_template(opts = {})
+      opts[:kind]               ||= "provision"
+      opts[:operatingsystem_id] ||= operatingsystem_id
+      opts[:hostgroup_id]       ||= hostgroup_id
+      opts[:environment_id]     ||= environment_id
+
+      ProvisioningTemplate.find_template opts
+    end
+
+    def available_template_kinds(provisioning = nil)
+      kinds = if provisioning == 'image'
+                cr     = ComputeResource.find_by_id(self.compute_resource_id)
+                images = cr.try(:images)
+                if images.blank?
+                  [TemplateKind.friendly.find('finish')]
+                else
+                  uuid       = self.compute_attributes[cr.image_param_name]
+                  image_kind = images.find_by_uuid(uuid).try(:user_data) ? 'user_data' : 'finish'
+                  [TemplateKind.friendly.find(image_kind)]
+                end
+              else
+                TemplateKind.all
+              end
+
+      kinds.map do |kind|
+        ProvisioningTemplate.find_template({ :kind               => kind.name,
+                                             :operatingsystem_id => operatingsystem_id,
+                                             :hostgroup_id       => hostgroup_id,
+                                             :environment_id     => environment_id
+        })
+      end.compact
+    end
+
+    def render_template(template)
```

Comment:
I'd prefer not mixing template selecting logic with rendering. Note that some plugins tend to include own extensions to `Host::Managed` but this would allow rendering in more classes. If it's out of scope for this PR, please keep rendering in host object.